sponsor-dapp-v2/src/lib/contracts.js

```diff
@@ -0,0 +1,32 @@
+import React, { useEffect } from "react";
+import { drizzleReactHooks } from "drizzle-react";
+
+function useDynamicallyAddedContract(contractAddress, abi) {
+  const { drizzle } = drizzleReactHooks.useDrizzle();
+  const { contract } = drizzleReactHooks.useDrizzleState(drizzleState => ({
+    contract: drizzleState.contracts[contractAddress]
+  }));
+
+  useEffect(() => {
```

Comment:
Could we limit how often this runs by adding an array of dependencies (maybe adding `contract`, `contractAddress`, and `abi` or something?

Reply:
Done.